pyrenew/convolve.py

```diff
@@ -217,3 +217,77 @@ def compute_delay_ascertained_incidence(
         mode="valid",
     )
     return delay_obs_incidence
+
+
+def daily_to_weekly(
+    daily_values: ArrayLike,
+    input_data_first_dow: int = 0,
+    week_start_dow: int = 0,
+):
+    """
+    Aggregate daily values (e.g.
+    incident hospital admissions) into weekly total values.
+
+    Parameters
+    ----------
+    daily_values : ArrayLike
+        Daily timeseries values (e.g. incident infections or incident ed visits).
+    input_data_first_dow : int
+        First day of the week in the input timeseries `daily_values`.
+        An integer between 0 and 6, inclusive (0 for Monday, 6 for Sunday).
+        If `input_data_first_dow` does not match `week_start_dow`, the incomplete first
+        week is ignored and weekly values starting
+        from the second week are returned. Defaults to 0.
+    week_start_dow : int
+        The desired starting day of the week for the output weekly aggregation.
+        An integer between 0 and 6, inclusive. Defaults to 0 (Monday).
+
+    Returns
+    -------
+    ArrayLike
+        Data converted to weekly values starting
+        with the first full week available.
+    """
+    if input_data_first_dow < 0 or input_data_first_dow > 6:
```

Comment:
In the future we may want to enforce integer-like values more strongly or conversely expand to support arrays of different potentially, but I think this is fine for now.